1.7.0: missing import for git_oidarray

Hi! I'm currently attempting the rebuild of projects against libgit2 1.7.0 on Arch Linux (https://archlinux.org/todo/libgit2-17/).

### Reproduction steps

Attempting to build recent jami-daemon against libgit2 (see https://git.jami.net/savoirfairelinux/jami-daemon/-/issues/876) fails due to a missing import for `git_oidarray`.
See recent [PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/jami-daemon/-/blob/6abb5dade3527c562e29adada2f613c568dc0897/PKGBUILD) for an overview on how we build jami-daemon.

### Expected behavior

The build succeeds.

### Actual behavior

Many instances of 

```
/usr/include/git2/sys/transport.h:116:17: error: ‘git_oidarray’ has not been declared
  116 |                 git_oidarray *out,
      |                 ^~~~~~~~~~~~
```

Full build log:
[jami-daemon-20230620-1-x86_64-build.log](https://github.com/libgit2/libgit2/files/12129406/jami-daemon-20230620-1-x86_64-build.log)


### Version of libgit2 (release number or SHA1)
1.7.0

### Operating system(s) tested
Arch Linux

--- a/include/git2/sys/transport.h
+++ b/include/git2/sys/transport.h
@@ -9,6 +9,7 @@
 #define INCLUDE_sys_git_transport_h
 
 #include "git2/net.h"
+#include "git2/oidarray.h"
 #include "git2/proxy.h"
 #include "git2/remote.h"
 #include "git2/strarray.h"

